resources/splunk_installer.rb

```diff
@@ -123,24 +127,28 @@ def download_package
     action :delete
   end
 
-  startup_files = if server? && systemd?
-                    [
-                      '/etc/systemd/system/splunk.service',
-                      '/etc/systemd/system/Splunkd.service',
-                    ]
-                  elsif systemd?
-                    [
-                      '/etc/systemd/system/splunk.service',
-                      '/etc/systemd/system/SplunkForwarder.service',
-                    ]
-                  else
-                    [ '/etc/init.d/splunk' ]
-                  end
-
-  (startup_files << cached_package).each do |f|
+  %w(splunk.service Splunkd.service SplunkForwarder.service).each do |name|
+    systemd_unit name do
+      action :delete
+      triggers_reload true
+    end
+  end
```

Comment:
Why are you removing the systemd unit files?

Reply:
This is in the `remove` action of the resource -- of course, the systemd unit files need to be removed.

Reply:
Doesn't the RPM/DEB package already handle that for us?

Reply:
It obviously does not, as the sequence in the recipe to remove the systemd unit files is executed AFTER the package is removed. Splunk's installer does not install the systemd unit files, except by a post-install command and it's entirely optional by the deployment environment. Been this way for years -- nothing new here, except improving the way that the files get removed.